### PR TITLE
Feat/pgroen/modbus rtu implementation

### PR DESCRIPTION
**Pull request by pgroen**
Implementation of the serial connection. Just a preliminary version.. Can be made very complex but for now we use it like this..

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,4 +8,6 @@ include(projectheader)
 project_header(modbus-cpp)
 
 add_subdirectory(src)
+add_subdirectory(tests)
+# add_subdirectory(tools)
```
Comment (TrueBrain) [inline CMakeLists.txt:12]: Que? :D

File: src/modbusrtu.cpp [hunk 2]
```diff
@@ -4,3 +4,96 @@
...
+
+ModbusRtu::ModbusRtu( const ConnectionConfig &conf )
+    : m_conConfig( conf )
+    , m_socket( -1 )
+{
+}
+
+bool ModbusRtu::Connect()
+{
+    m_socket = open( m_conConfig.getPortName().c_str(), O_RDWR);
```
Comment (TrueBrain) [inline src/modbusrtu.cpp:28]: Shall I nitpick about the wide variance in whitespace usage, or shall I just ignore that? :P

(space here between `(` and `m_conConfig`, line 29 has no spaces around `()`, line 37 has, etc etc :P

File: src/modbusrtu.cpp [hunk 3]
```diff
@@ -4,3 +4,96 @@
...
+    {
+        return false;
+    }
+
+    struct termios l_tty;
+
+    // Read in existing settings, and handle any error
+    if( tcgetattr(m_socket, &l_tty) != 0 )
+    {
+        // Replace later on with a logger line.
```
Comment (TrueBrain) [inline src/modbusrtu.cpp:39]: Maybe mark it with TODO, so you can actually find back these statements?

File: tools/modbus-slave/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+set(SRC_LIST
+    ${CMAKE_CURRENT_SOURCE_DIR}/modbusbase.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/modbusbase.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/modbustcp.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/modbustcp.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/modbusrtu.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/modbusrtu.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/connectionconfig.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/connectionconfig.cpp
```
Comment (TrueBrain) [inline tools/modbus-slave/CMakeLists.txt:21]: Bit weird to have this in the tools folder, while it includes the `src` files? No clue what the actual intention is of these files .. `tools`.. *shrug* :D

File: src/modbusrtu.h [hunk 5]
```diff
@@ -6,14 +6,39 @@
...
 
+    virtual bool Connect() override;
+    virtual bool Close() override;
+
+    /*!
+     * \brief modbusSend
+     * \param to_send
+     * \param length
+     * \return
+     */
```
Comment (TrueBrain) [inline src/modbusrtu.h:29]: Basically, these are empty doxygen comments :P

File: src/modbusrtu.cpp [hunk 6]
```diff
@@ -4,3 +4,96 @@
...
+        return false;
+    }
+
+    /* Control modes */
+    l_tty.c_cflag &= ~PARENB;                   // Clear parity bit, disabling parity (most common)
```
Comment (TrueBrain) [inline src/modbusrtu.cpp:44]: Of course this should be configurable, from the `m_conConfig` object. The information is already in there, you should just get it out :) Also holds for other fields :)

File: src/modbusrtu.cpp [hunk 7]
```diff
@@ -4,3 +4,96 @@
...
+    l_tty.c_cflag &= ~CSTOPB;                   // Clear stop field, only one stop bit used in communication (most common)
+    l_tty.c_cflag &= ~CSIZE;                    // Clear all bits that set the data field
+    l_tty.c_cflag |= CS8;                       // 8 bits per byte (most common)
```
Comment (TrueBrain) [inline src/modbusrtu.cpp:47]: The comment here is misleading. `most common` is not true. For RTU, this is, by definition, 8 bits. 7 bits would be ASCII.

**Review by TrueBrain — APPROVED** [review]
You got to love the simplicity of using POSIX things :) So little code ..

Does it also actually work? I couldn't test that over here :)

File: CMakeLists.txt (hunk 1 above)
Comment (pgroen) [reply CMakeLists.txt:12]: Already preparing for implementation of test-tools. Interactive configuration and communication

File: src/modbusrtu.cpp (hunk 2 above)
Comment (pgroen) [reply src/modbusrtu.cpp:28]: I see what you mean. Force of habit..

File: src/modbusrtu.cpp (hunk 3 above)
Comment (pgroen) [reply src/modbusrtu.cpp:39]: Ghe ghe!

File: tools/modbus-slave/CMakeLists.txt (hunk 4 above)
Comment (pgroen) [reply tools/modbus-slave/CMakeLists.txt:21]: Still a WIP... Didn't figure this one out / thought this through yet...

**Pull request merged**